cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java

```diff
@@ -541,4 +547,20 @@ public static void sanityCheckNonExistingGoal(List<String> goals, Map<String, Go
 
     return balancednessCostByGoal;
   }
+
+  /**
+   * Reads the configuration file, parses and validates the configs.
```

Comment:
Should we also mention the properties that enable configs to be provided out of the configuration file? 
Is it intentional to override these hardcoded properties if they were also defined in the configuration file?

Reply:
1st: acked, will do
2nd: makes sense, our requirements were quite simple but I can agree that people may want to do something else so it isn't optimal to override their configs.

Reply:
1st: acked, will do
2nd: well, I treat them as defaults here so if people want to do something else than what CC does by default, then they should be free to. What can I improve here?